### PR TITLE
DOC: Hardware Compatibility / Sensors edits

### PR DESCRIPTION
**Pull request by dbkinder**
Added organization to the new hardware compatibilty documents and
edited the documents submitted for the wireless-networking
area.  The intent is to organize the HW compatibility docs into logical
groups based on their functionality, with separate docs for each of
the peripherals.  These separate docs would explain details about that module
(as was done for the two Zigbee modules).

at86rf222.rst
cc2520.rst
  These two recently submitted docs were moved down a level in the
  hardware folder hierarchy, along with images they reference.

index.rst
  Added the hardware compability section to the overall documentation index page.

[skip ci]

Signed-off-by: David Kinder <david.b.kinder@intel.com>

### CODE REVIEW

File: doc/hardware/wireless-networking.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+.. _wireless-networking:
+
+Wireless Networking
+###################
+
+Drivers for these wireless networking cards are available for the Ostro |trade| OS and have been tested to work
+on :ref:`platforms` as indicated within the following device-specific documents. 
+
+
+ZigBee / IEEE 802.15.4
```
Comment (mythi) [inline doc/hardware/wireless-networking.rst]: Let's drop Zigbee

File: doc/hardware/wireless-networking/at86rf212.rst [hunk 2]
```diff
@@ -0,0 +1,115 @@
+.. _at86rf212:
+
+AT86RF212 - 700/800/900 MHz ZigBee/IEEE 802.15.4 RF transceiver
+###############################################################
+
+| Category: radio
```
Comment (mythi) [inline doc/hardware/wireless-networking/at86rf212.rst]: "wireless-networking"?

**Comment by dbkinder** [comment]
Force push updated to address Mikko's comments

**Comment by mythi** [comment]
+1

**Pull request merged**